tests/phpunit/CRM/Civicase/Service/CaseCategoryCustomFieldExtendsTest.php

```diff
@@ -0,0 +1,146 @@
+
+    $caseCategoryCustomFieldExtends->create(
+      $entityValue,
+      $label
+    );
+
+    $customFieldCreated = $this->getCustomFieldByValue($entityValue);
+    $this->assertEquals(1, $customFieldCreated['count']);
+    $this->assertEquals($customFieldCreated['values'][0]['value'], $entityValue);
+    $this->assertEquals($customFieldCreated['values'][0]['label'], $label);
+  }
+
+  /**
+   * Test that calling twice the create method does not produce an error.
+   */
+  public function testIsNotCreatedTwice() {
+    $entityValue = 'Test Entity Value';
+    $label = 'Test Label';
+    $entityTypeFunction = 'Test Entity Type Function';
+    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
+
+    // First call.
+    $caseCategoryCustomFieldExtends->create($entityValue, $label, $entityTypeFunction);
+    // Second call.
+    $caseCategoryCustomFieldExtends->create($entityValue, $label, $entityTypeFunction);
+
+    $customFieldCreated = $this->getCustomFieldByValue($entityValue);
+    $this->assertEquals(1, $customFieldCreated['count']);
+  }
+
+  /**
+   * Test deleting an existing custom field.
+   */
+  public function testIsSuccessfullyDeleted() {
+    // Create a random instance of custom field.
+    $caseTypeCategory = $this->createRandomCustomField();
+    // We verify that is correctly stored.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(1, $customField['count']);
+
+    // Delete the custom field.
+    (new CaseCategoryCustomFieldExtends())->delete($caseTypeCategory['value']);
+
+    // And verify is not counted anymore.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(0, $customField['count']);
+  }
+
+  /**
+   * Test that calling twice the delete method does not produce an error.
+   */
+  public function testDeleteCanBeCalledTwice() {
+    // Create a random instance of custom field.
+    $caseTypeCategory = $this->createRandomCustomField();
+    // We verify that is correctly stored.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(1, $customField['count']);
+
+    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
+    // First call to delete method.
+    $caseCategoryCustomFieldExtends->delete($caseTypeCategory['value']);
+    // Second call to delete method.
+    $caseCategoryCustomFieldExtends->delete($caseTypeCategory['value']);
+
+    // And verify is not counted anymore.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(0, $customField['count']);
+  }
+
+  /**
+   * Returns a custom field by it value.
+   *
+   * @param string $value
+   *   Value of the custom field extends.
+   *
+   * @return array
+   *   Custom Field Extends details
+   */
+  private function getCustomFieldByValue($value) {
```

Comment:
Can we rename this function. This does not actually return a custom field but returns an option value belonging to the 'cg_extend_objects' option group. Maybe rename to `getOptionValueForCgExtendsGroup` or something better.